### PR TITLE
Feature/2875 post cardview

### LINKED ISSUE #2875: CardView for posts/pages
The Calypsofication of posts/pages requires switching to a CardView (more easily done now that #2802 is merged), which involves:

* Featured image display above the post (similar to the reader)
* Buttons for Edit / View / Stats - edit shows the post editor, view shows the published post in a webView, stats shows the stats detail screen for that post
* ~~Only Me / Everyone switch for multi-author blogs~~

This is the largest - and primary - task for the Calypsofication effort.

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #2875 - This PR represents phase 1 of the post/pages redesign. Apologies for the size - I tried to minimize it by splitting off several subtasks into separate PRs, but so many changes were needed to the underlying code that it still turned out to be quite large.

Changes in this PR include:

* Posts/pages are shown in a card view that includes an excerpt and featured image (caveat: see #2932)
* Each card has a toolbar-ish area beneath it (similar to that in the current Calypso post list) which enables editing/viewing/trashing the post as well as viewing its stats
* Trashing a post (or deleting a local draft) uses an undo bar rather than dialog confirmation

**Note:** For now tapping a post opens it in the editor, which isn't a great experience. This will be changed in #2933 to open the post in a new preview activity.

device-2015-07-01-081953

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java [hunk 1]
```diff
@@ -72,11 +73,29 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    public void onPause() {
+        super.onPause();
+        AniUtils.showFab(mFabView, false);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: Much more appropriate to show/hide the FAB in Pause/Resume methods. :+1:

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderItemDecoration.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package org.wordpress.android.ui.reader.views;
+
+import android.graphics.Rect;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+/**
+ * dividers for reader cards
+ */
+public class ReaderItemDecoration extends RecyclerView.ItemDecoration {
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderItemDecoration.java]: Should this be renamed (and move packages) to something more generic now that it's used in both Reader and Post lists?

File: WordPress/src/main/java/org/wordpress/android/WordPressDB.java [hunk 3]
```diff
@@ -1010,37 +1011,70 @@ public void savePosts(List<?> postsList, int localBlogId, boolean isPage, boolea
...
+            return listPosts;
+        } finally {
+            SqlUtils.closeCursor(c);
+        }
+    }
+
+    private Post getPostFromCursor(Cursor c) {
+        Post post = new Post();
+
+        post.setLocalTableBlogId(c.getInt(c.getColumnIndex("blogID")));
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/WordPressDB.java]: It'd be nice to have these column names defined somewhere (like [this](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/WordPressDB.java#L56)). Probably best for another PR though and not really in scope here since you're just moving these lines.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderItemDecoration.java (hunk 2 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderItemDecoration.java]: Agreed. Fixed in 68be559.

File: WordPress/src/main/java/org/wordpress/android/WordPressDB.java (hunk 3 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/WordPressDB.java]: Agreed, both that this should be done and that it should be done in another PR ;)

**Comment by tonyr59h** [comment]
There's been plenty of discussion around merging posts/pages classes. It's good to see it happening here, good work!

**Comment by tonyr59h** [comment]
It looks like we lost the no network toast when refreshing the Post/Pages lists.

**Comment by tonyr59h** [comment]
Ahh, just read the [commit message](https://github.com/wordpress-mobile/WordPress-Android/commit/baa4fa0bf69c19324cde7f8a05169612138fbde0). It makes sense to remove it if the empty view is showing, but if there are items in the list and no network I'm not receiving an indication that the refresh failed.

**Comment by nbradbury** [comment]
> It looks like we lost the no network toast when refreshing the Post/Pages lists.

I confess to being over-zealous in my toast removal, since the existing code showed toasts far too often. I think the only place we should have a toast now is after a manual refresh (PTR) fails. I can add that to this PR or as a separate issue - your call.

**Comment by tonyr59h** [comment]
Sometimes selecting UNDO after trashing a post will not succeed and the UNDO button will disappear without the post returning. Refreshing doesn't bring it back so I suspect the trashed state is still set. Seems to happen when I select UNDO, trash it again quickly, and select UNDO again.

**Comment by tonyr59h** [comment]
> the existing code showed toasts far too often

Agreed, and it's usually not the best way to handle an error. I'll file an issue for it so we can tackle it in another PR.

**Comment by tonyr59h** [comment]
Another fun one with the UNDO button. If you delete a post then rotate it comes back on rotation (probably due to refresh).

**Comment by nbradbury** [comment]
> Another fun one with the UNDO button. If you delete a post then rotate it comes back on rotation

Should be fixed by 5424527.

**Comment by nbradbury** [comment]
> Seems to happen when I select UNDO, trash it again quickly, and select UNDO again.

This was caused by using a single variable to determine whether a post was trashed. Fixed in 65a0208 by using a list of trashed posts instead.

**Comment by nbradbury** [comment]
@tonyr59h Ready for round two.
